web-common/src/features/connectors/olap/test-connection.ts

```diff
@@ -0,0 +1,66 @@
+import { get } from "svelte/store";
+import { queryClient } from "../../../lib/svelte-query/globalQueryClient";
+import { waitUntil } from "../../../lib/waitUtils";
+import {
+  connectorServiceOLAPListTables,
+  getConnectorServiceOLAPListTablesQueryKey,
+} from "../../../runtime-client";
+import { fileArtifacts } from "../../entity-management/file-artifacts";
+import { humanReadableErrorMessage } from "../../sources/modal/errors";
+
+interface TestConnectorResult {
+  success: boolean;
+  error?: string;
+}
+
+export async function testOLAPConnector(
```

Comment:
A method called `testOLAPConnector` seems to be also checking if the file has updated as well.

Perhaps we should split them out, update and confirming of reconcile can be in a method. We could also get rid of the hard wait for `1000s`. We could check for version change in ProjectParser instead?

Reply:
Ya this is better!